src/status_im/contexts/wallet/send/input_amount/events.cljs

```diff
@@ -0,0 +1,36 @@
+(ns status-im.contexts.wallet.send.input-amount.events
+  (:require
+    [status-im.contexts.wallet.send.input-amount.controlled-input-logic :as controlled-input-logic]
+    [utils.re-frame :as rf]))
+
+(def input-value-path [:wallet :ui :screens :input-amount-screen :input-value])
```

Comment:
@ilmotta A good point to discuss in our DX for sure.

I don't agree on that this pattern is inefficient, if subs are properly structured, I think this is the way re-frame is optimized to work: through a graph (tree) of signals (subs) and mounting only the subs being used by views.

Declaring everything as root values and compose keywords for every value stored sounds, to me, as a wrong pattern for re-frame :thinking: 

In terms of performance, if we store everything as root subs, each change in app-db will trigger a comparison on every stored value. This giant amount of comparisons won't happen if we nest values and write subs for the subnodes. So I don't see why it'd be worse to nest data. - We already discussed this in Montenegro and I remember an experienced dev said they thought re-frame compared by checking references, but it's not the case.

> It's also a bit of a smell that we are naming state with UI concerns, like input-amount-screen. Treating the app-db like a database of sorts seems better, at least in my experience.

I've seen both approaches in production, I personally prefer not lift up component's state to re-frame, but it's not that bad, as long as it's properly implemented (e.g. by using `dispatch-sync` when handling inputs).

I agree with you on this point, and, to me, status doesn't use this pattern of handling the UI state in re-frame. I think good results can be achieved by using reagent and it's tools (like cursors).

Reply:
> Declaring everything as root values and compose keywords for every value stored sounds, to me, as a wrong pattern for re-frame 🤔

@ulisesmac, not everything. I wouldn't suggest always using root keys, of course there's a balance and only simple extractors should be closer to the root.

I think until a use case is built the discussion won't advance much. If I could find some time I'd love to dive deeper to make the discussion more practical, but unfortunately I don't have much else to say.

For now, my intuition and a some experience tells me that modeling re-frame state is not much different than modeling a database and that a database that's not reasonably well normalized is quite hard to optimize/change. Normalization is generally good for many reasons. There's a reason people suffered with the likes of MongoDB or Firestore when nesting collections and couldn't easily write efficient queries, to give an example.

Reply:
> There's a reason people suffered with the likes of MongoDB or Firestore when nesting collections and couldn't easily write efficient queries, to give an example.

I completely agree since I've used both.

> I think until a use case is built the discussion won't advance much. 

Yeah, having real examples of storing and querying data with both approaches compared would be great to properly compare what's the best for us.